File: src/cli/test/logger_test.py
import logging
import openstudio
import sys

# Root logger
logger = logging.getLogger()
logger.setLevel(logging.WARNING)

formatter = logging.Formatter('LOGGER - %(message)s')
handler = logging.StreamHandler(sys.stdout)
handler.setLevel(logging.WARNING)
handler.setFormatter(formatter)
logger.addHandler(handler)

openstudio.Logger_instance().standardOutLogger().setLogLevel(openstudio.Error)

logger.info("STDOUT Info")
logger.warning("STDOUT Warn")
logger.error("STDOUT Error")

openstudio.logFree(openstudio.Info, "test", "Info")
openstudio.logFree(openstudio.Warn, "test", "Warn")
openstudio.logFree(openstudio.Error, "test", "Error")

File: src/cli/test/logger_test.rb
require 'logger'

$logger = Logger.new(STDOUT)
$logger.level = Logger::WARN
$logger.formatter = proc do |severity, datetime, progname, msg|
   "LOGGER - #{msg}\n"
end

OpenStudio::Logger.instance.standardOutLogger.setLogLevel(OpenStudio::Error)

$logger.info "STDOUT Info"
$logger.warn "STDOUT Warn"
$logger.error "STDOUT Error"

OpenStudio::logFree(OpenStudio::Info, "test", "Info")
OpenStudio::logFree(OpenStudio::Warn, "test", "Warn")
OpenStudio::logFree(OpenStudio::Error, "test", "Error")
